### ISSUE TITLE
Support new `pydantic_v2` manifest and JSON Schemas

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-web#531]
As already explored/discussed in https://github.com/fractal-analytics-platform/fractal-web/issues/479, we are currently not supporting the `anyOf` pattern for nullable JSON Schema properties - although we may want to do it in the future. Thus are working in `fractal-tasks-core` to move to Pydantic V2 while keeping the generated manifest as close as possible to the V1 one (see e.g. https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/801).

Here is an example of the old/new manifests for `fractal-tasks-core` (perhaps up to some last-minute additions):

Manifest with Pydantic V1: https://raw.githubusercontent.com/fractal-analytics-platform/fractal-tasks-core/f40116aedd211dac4a1b6d3eea3dbbb7734b1e86/fractal_tasks_core/__FRACTAL_MANIFEST__.json

Manifest with Pydantic v2: [manifest_pydantic_v2.json](https://github.com/user-attachments/files/16310006/manifest_pydantic_v2.json)

We should review the remaining differences, and handle them in fractal-web.

One that I already noticed is that `"definitions"` is now replaced by `"$defs"`. I'll mention here any other change that I find.

**Comment by tcompa** [comment]
I took one of the new manifests, and I replaced `$defs` with `definitions` by hand (`sed -i -e 's/$defs/definitions/g' fractal_tasks_core/__FRACTAL_MANIFEST__.json`). The remaining diff seems quite harmless:
```diff

**Comment by tcompa** [comment]
TL;DR

The only relevant differences are
1. The `definitions` renaming
2. Sometimes we have a dummy single-element `allOf`, which should be already supported (https://github.com/fractal-analytics-platform/fractal-web/issues/413)

@zonia3000 if all seems reasonable to you, then the required change to fractal-web is just to accept both `pydantic_v1` and `pydantic_v2`.

**Comment by zonia3000** [comment]
@tcompa  In the Add Single task component, when an args schema file is selected, we are displaying an "Args schema version" input having `pydantic_v1` as default value. Should I set the default to `pydantic_v2`?

![image](https://github.com/user-attachments/assets/d83497dd-b2de-4dd1-9867-366a605c9525)

**Comment by tcompa** [comment]
> In the Add Single task component, when an args schema file is selected, we are displaying an "Args schema version" input having pydantic_v1 as default value. Should I set the default to pydantic_v2?

Yes, this seems the correct approach to me.

For the record, the `args_schema_version` attribute was introduce only to be future-proof in fractal-web. It is not used or validated in any other Fractal component, as far as I remember. Therefore as soon as fractal-web supports both strings, it is fine to always default to the latest.
